activerecord/lib/active_record/model_schema.rb

```diff
@@ -271,6 +256,18 @@ def inheritance_column=(value)
         @explicit_inheritance_column = true
       end
 
+      # The list of columns names the model should ignore. Ignored columns won't have attribute
+      # accessors defined, and won't be referenced in SQL queries.
+      def ignored_columns
+        @ignored_columns ||= [].freeze
```

Comment:
We need to delegate to `superclass.ignored_columns` to keep the current behavior like here:

```diff
--- a/activerecord/lib/active_record/model_schema.rb
+++ b/activerecord/lib/active_record/model_schema.rb
@@ -126,6 +126,7 @@ module ModelSchema
       class_attribute :pluralize_table_names, instance_writer: false, default: true
 
       self.inheritance_column = "type"
+      self.ignored_columns = [].freeze
 
       delegate :type_for_attribute, to: :class
 
@@ -259,7 +260,11 @@ def inheritance_column=(value)
       # The list of columns names the model should ignore. Ignored columns won't have attribute
       # accessors defined, and won't be referenced in SQL queries.
       def ignored_columns
-        @ignored_columns ||= [].freeze
+        if defined?(@ignored_columns)
+          @ignored_columns
+        else
+          superclass.ignored_columns
+        end
       end
 
       # Sets the columns names the model should ignore. Ignored columns won't have attribute
```

Reply:
Thank you for the concrete code. The delegation is certainly necessary.